cirq/ops/projectors_test.py

```diff
@@ -0,0 +1,114 @@
+import re
+
+import numpy as np
+import pytest
+
+import cirq
+
+
+def test_projector_circuit_diagram():
+    q = cirq.NamedQubit('q')
+    projector = cirq.Projector([[1.0, 0.0]])
+    cirq.testing.assert_has_diagram(cirq.Circuit(projector(q)),
+                                    "q: ───Proj([[1.0, 0.0]])───",
+                                    precision=None)
+    cirq.testing.assert_has_diagram(cirq.Circuit(projector(q)),
+                                    "q: ───Proj([[1.0, 0.0]])───",
+                                    precision=2)
+
+
+def test_projector_qubit():
+    zero_projector = cirq.Projector([[1.0, 0.0]])
+    one_projector = cirq.Projector([[0.0, 1.0]])
+
+    np.testing.assert_allclose(cirq.channel(zero_projector),
+                               ([[1.0, 0.0], [0.0, 0.0]],))
+
+    np.testing.assert_allclose(cirq.channel(one_projector),
+                               ([[0.0, 0.0], [0.0, 1.0]],))
+
+
+def test_projector_from_np_array():
+    zero_projector = cirq.Projector(np.array([[1.0, 0.0]]))
+    np.testing.assert_allclose(cirq.channel(zero_projector),
+                               ([[1.0, 0.0], [0.0, 0.0]],))
+
+
+def test_projector_bad_rank():
+    with pytest.raises(ValueError,
+                       match="The input projection_basis must be a 2D array"):
+        cirq.Projector(np.array([1.0, 0.0]))
+
+
+def test_projector_plus():
+    plus_projector = cirq.Projector([[1.0, 1.0]])
+
+    np.testing.assert_allclose(cirq.channel(plus_projector),
+                               ([[0.5, 0.5], [0.5, 0.5]],))
+
+
+def test_projector_overcomplete_basis():
+    with pytest.raises(ValueError,
+                       match="Vectors in basis must be linearly independent"):
+        cirq.Projector([[1.0, 0.0], [0.0, 1.0], [1.0, 1.0]])
+
+
+def test_projector_non_orthogonal_basis():
+    cirq.Projector([[1.0, 0.0]], enforce_orthonormal_basis=True)
+    cirq.Projector([[1.0, 0.0], [0.0, 1.0]], enforce_orthonormal_basis=True)
+
+    with pytest.raises(ValueError, match="The basis must be orthogonal"):
```

Comment:
I would add another test case: to test for the bases being of norm 1.  
You are testing against both norm 1 and orthogonality here, maybe it's worthwhile separating out the two cases.

Reply:
Done.